src/Turtle/Prelude.hs

```diff
@@ -1090,6 +1093,32 @@ time io = do
 hostname :: MonadIO io => io Text
 hostname = liftIO (fmap Text.pack getHostName)
 
+-- | Show the full path of an executable file
+which :: MonadIO io => FilePath -> io (Maybe FilePath)
+which cmd = fold go Control.Foldl.head
```

Comment:
I don't feel strongly about it either, but the list behavior is closer to `which --all`, if that's worth anything.